### PR TITLE
[Move stdlib] add test to ensure that generated files are up to date

### PR DESCRIPTION
**Pull request by matbd**
## Motivation

Run the stdlib tool in CI and use the git state to make sure all generated files are up to date.

Using git is not ideal but at least we err on the safe side.

## Test Plan

CI

### CODE REVIEW

**Comment by bmwill** [comment]
What's the intent of this test? We already have a check in CI for this so idk how much more benefit this would give us

**Comment by matbd** [comment]
> What's the intent of this test? We already have a check in CI for this so idk how much more benefit this would give us

@bmwill Cool, do you have a link? It must be somehow recent (or partial?) because in the past I noticed that running the stdlib tool would sometimes generate more changes than my PR. (Also, we were not necessary deleting obsolete generated files, see #4600)

**Comment by matbd** [comment]
Also, https://github.com/libra/libra/pull/5319 just landed with a typo breaking the stdlib tool.

**Comment by bmwill** [comment]
This script is run as a part of CI: https://github.com/libra/libra/blob/master/scripts/changed-files.sh

Though I now understand how this diff is different. This actually runs that binary and verifies theres no changed files.

**Comment by matbd** [comment]
> This script is run as a part of CI: https://github.com/libra/libra/blob/master/scripts/changed-files.sh
> 
> Though I now understand how this diff is different. This actually runs that binary and verifies theres no changed files.

Oh I see. I'm totally going to steal the `--porcelain` trick.

There is definitely some overlapping but as you said, the new script runs the tool (and it is not guaranteed to be run at the end of CI).

**Comment by jolestar** [comment]
How about using build.rs for auto compile stdlib at the build stage?

**Comment by matbd** [comment]
> How about using build.rs for auto compile stdlib at the build stage?

We want to make it easy to use the generated objects, including from outside the repo.

File: language/transaction-builder/generator/src/rust.rs [hunk 1]
```diff
@@ -52,7 +52,7 @@ fn output_preamble(out: &mut dyn Write, local_types: bool) -> Result<()> {
 
 // This file was generated. Do not modify!
 //
-// To re-generate this code, run: `(cd language/stdlib && cargo run --release)`
```
Comment (phlip9) [inline language/transaction-builder/generator/src/rust.rs:55]: does stdlib tool now work outside of language/stdlib? it breaks for me otherwise (though my branch is several days off master)

Comment (matbd) [reply language/transaction-builder/generator/src/rust.rs:55]: I thought I tried but let me try again (and change the cwd in the test command)

Comment (matbd) [reply language/transaction-builder/generator/src/rust.rs:55]: problem solved
